text/000-sort-variable-unification.md

```diff
@@ -0,0 +1,77 @@
+- Title: Introducing quality variables in sort unification
+
+- Drivers: Pierre-Marie Pédrot
+
+----
+
+# Summary
+
+In addition to variables for universe levels, we generalize the type of sorts to
+also contain quality variables, allowing to delay the choice between Prop, Type
+and SProp made by the unification. This CEP is only concerned about unification,
+although the kernel structures need to be modified accordingly, the kernel
+treats undefined quality variables as an error, similarly to evars.
+
+The first part of this CEP is implemented in [Coq PR 16903](https://github.com/coq/coq/pull/16903).
+
+# Motivation
+
+There are several underlying motivations for this CEP.
+
+- Immediate fix: alleviate the longstanding issue of having to explicitly
+annotate variables with `Prop` to make the unification go through.
```

Comment:
Is this going to force `Prop` anytime you accidentally or intentionally use impredicativity?  e.g., is `Check let S := _ in let T : S := _ in (forall A : S, T) : S.` expected to succeed, instantiating `?S` to `Prop`?

Reply:
No, both for usability and completeness. There are several impredicative universes so there is no reason to pick one randomly.